### DISCUSSION TITLE
Is it possible to combine filters for different mutations in useIsMutating hook

### DISCUSSION THREAD

**Discussion by JohnnyJumper** [TanStack/query#3426] [Q&A] [answered]
I have various mutations across many different states and I wanted to utilize the useIsMutating with the specific mutation keys to know exactly when the needed mutation is in the mutating state. 

At the moment the only way I could make it work is the following:

```
export const useCartIsMutating = () => {
  const isAddingToCart = useIsMutating(queryKey.addToCart()) > 0;
  const isRemovingFromCart = useIsMutating(queryKey.removeFromCart()) > 0;
  const isClearingCart = useIsMutating(queryKey.clearCart()) > 0;
  return isAddingToCart || isRemovingFromCart || isClearingCart;
};
``` 

Is it possible to combine all those useIsMutating with one function call? I tried the following setups and none of them worked for me so far:

```
useIsMutating([
{ mutationKey: queryKey.addToCart() },
{ mutationKey: queryKey.removeFromCart() },
{ mutationKey: queryKey.clearCart() }
```

```
useIsMutating([
  queryKey.addToCart(),
  queryKey.removeFromCart(),
  queryKey.clearCart() 
]);
```

```
useIsMutating( [
[ queryKey.addToCart() ],
[ queryKey.removeFromCart() ],
[ queryKey.clearCart() ]
);
```

Would be great to know if there is a way to make it happens somehow as none of the three variants above worked for me and all of them are always returning 0.

If this is a valid idea and there is no implementation I would love to contribute and make a PR for this feature. Please advice :)

**Comment by TkDodo** [comment, marked as answer]
you can try `useIsMutating({ predicate: (mutation) => mutation.key === ...  || mutation.key === ... })`

with `predicate`, you can return true or false for every matching mutation. If your keys are known and you don't need fuzzy matching of keys, you should be able to combine them.

**Reply by JohnnyJumper** [reply]
Thank you! I kinda overlooked the predicate option. This works great for me!
